### PR TITLE
Enable revised HSL in single-coin GPU optimization

### PR DESCRIPTION
**Pull request by enarjord**
Enable `live.hsl_engine=revised` in single-coin GPU optimization for both strategy families and all three signal modes. Canonical effective policies reach the revised shader explicitly; unified HSL uses portfolio bounds and one shared policy instead of directional legacy defaults. Exact Rust validation remains authoritative for retained candidates, and checkpoint identity includes the revised engine and policies.

Revised GPU scenarios require 1m candles and exactly one prepared coin. Multi-coin scenarios remain explicitly rejected pending their controller and scratch-memory integration. Legacy policy transport and checkpoint shape remain unchanged.

Validation:
- Actual Metal and CUDA: canonical service execution for all three modes and both strategies, override precedence, and public offline optimizer runs with exact Rust evaluation and checkpoint resume in all three modes.
- Existing GPU backend/service and revised optimizer contract tests pass; explicit unsupported-scope and documentation tests pass.
- Source-verified native runtime on both platforms. No live exchange execution.

### CODE REVIEW

File: src/optimization/gpu/service.py [hunk 1]
```diff
@@ -2111,6 +2117,10 @@ def __init__(
             signal_mode=signal_mode,
             hsl_enabled=bool(hsl_enabled_sides),
         )
+        if self.hsl_engine == "revised":
+            if candle_interval_minutes != 1:
+                raise ValueError("Revised GPU HSL requires 1m candles")
+            pnl_lookback_bars = int(float(backtest_params["pnls_max_lookback_days"]) * 1440)
```
[the remark this answers is not in the document]
Comment (enarjord) [reply src/optimization/gpu/service.py]: The disabled-policy sentinel case is confirmed and will be fixed with service/runner regressions. One part needs reconsideration: revised HSL does not use the legacy `pnl_lookback_bars` field. `backtest_hsl_revised.rs` computes `start = now - round(days * 86_400_000)` and clips fills/observations against that timestamp; `backtest_hsl_cache.rs` uses the same boundary. For minute-aligned observations, a 1440.25-minute duration retains 1440 minutes of history, not 1441. Using ceil would introduce a revised mismatch.

I will transport the rounded millisecond boundary to an integer-minute cutoff and test fractional-minute and sub-millisecond edge cases. Please reconsider the requested ceil conversion against the revised input builder; the inactive sentinel repair remains accepted.

**Comment by enarjord** [comment]
@codex review

Please review the current head, including both previous findings. Disabled revised policies now keep the zero/all history sentinel inert through both strategy runners. Coin-mode HSL search activity uses effective policies for actual prepared members, including suite membership and market-source identity. New regressions and actual Metal/CUDA optimizer, exact-validation and resume runs pass.

The first thread contains a source-backed request to reconsider the proposed legacy ceil conversion: revised Rust clips using a rounded millisecond timestamp, so the minute-aligned proxy now derives its cutoff from that same rounding. Fractional-minute and sub-millisecond edges are covered. Please explicitly clear or revisit that distinction; existing threads remain unresolved pending this review.

File: src/optimization/backends/gpu_backend.py [hunk 2]
```diff
@@ -3588,21 +3617,23 @@ def _gpu_hsl_search_sides(
...
     target_sides = {
-        side
-        for side in ("long", "short")
-        if any(
-            gpu_side_enabled(item, side)
-            and _gpu_hsl_side_enabled(item, side)
-            for item in configs
-        )
+        side for side in ("long", "short")
+        if any(gpu_side_enabled(item, side) and _gpu_hsl_side_enabled(item, side)
```
[the remark this answers is not in the document]
Comment (enarjord) [reply src/optimization/backends/gpu_backend.py]: Addressed in 7728f9013: revised coin-mode activity now reuses the canonical effective-policy resolver with actual prepared coin membership and market-source metadata. Added enabled-base/disabled-member, disabled-base/enabled-member, out-of-dataset override, and narrower suite-membership regressions. Keeping this thread open until current-head review clears it.

**Comment by enarjord** [comment]
@codex review

Please review the current head, including both previous findings. Disabled revised policies now keep the zero/all history sentinel inert through both strategy runners. Coin-mode HSL search activity uses effective policies for actual prepared members, including suite membership and market-source identity. New regressions and actual Metal/CUDA optimizer, exact-validation and resume runs pass.

The first thread contains a source-backed request to reconsider the proposed legacy ceil conversion: revised Rust clips using a rounded millisecond timestamp, so the minute-aligned proxy now derives its cutoff from that same rounding. Fractional-minute and sub-millisecond edges are covered. Please explicitly clear or revisit that distinction; existing threads remain unresolved pending this review.
